compose-ktx/src/main/kotlin/modifier/ApplyIf.kt

```diff
@@ -0,0 +1,32 @@
+package com.redmadrobot.extensions.compose.modifier
+
+import androidx.compose.ui.Modifier
+
+/**
+ * Applies the given block of modifications to the Modifier if the condition is true.
+ *
+ * @param condition condition to determine if the block should be applied.
+ * @param block Lambda function that modifies the Modifier.
+ * @return Modified Modifier based on the condition.
+ */
+public inline fun Modifier.applyIf(
+    condition: Boolean,
+    block: Modifier.() -> Modifier,
+): Modifier = applyChoice(condition = condition, trueBlock = block, falseBlock = { this })
```

Comment:
tried.

I can't tell the compiler in any way that we are checking the type for null inside a boolean expression.

The first thing that came to mind:

```kotlin
public inline fun <T> T.applyIf(
    condition: Boolean,
    block: T.() -> T,
): T = applyIfElse(condition = condition, trueBlock = block, falseBlock = { this })

public inline fun <T> T.applyIfElse(
    condition: Boolean,
    trueBlock: T.() -> T,
    falseBlock: T.() -> T,
): T {
    return if (condition) {
        require(condition) // <-
        trueBlock()
    } else {
        falseBlock()
    }
}
```

And then, already a little desperate: 

```kotlin
private fun check() {
    val x: Int? = 5
    var y: Int = 3

    // doesn't want to work
    y.applyIf(x != null) {
        // compile error Type mismatch.
        y += x
        y
    }

    // work
    run {
        require(x != null)
        y += x
    }
}

public inline fun <T> T.applyIf(
    condition: Boolean,
    block: T.() -> T,
): T = applyIfElse(condition = condition, trueBlock = block, falseBlock = { this })

public inline fun <T> T.applyIfElse(
    condition: Boolean,
    trueBlock: T.() -> T,
    falseBlock: T.() -> T,
): T {
    return if (condition) {
        runTrue(condition, trueBlock)
    } else {
        falseBlock()
    }
}

@OptIn(ExperimentalContracts::class)
public inline fun <T> T.runTrue(
    condition: Boolean,
    trueBlock: T.() -> T
): T {
    contract {
        returns() implies condition
    }

    return if (condition) trueBlock() else throw IllegalStateException("message".toString())
}
```

But it also seems impossible because this one doesn't work either:

```kotlin

    // work
    run {
        require(x != null)
        y += x
    }


    // doesn't
    run {
        require( {x != null}.invoke() )
        // compile error Type mismatch.
        y += x
    }


```

Reply:
That's sad. It looks like the only possible way to do it is to introduce a new function, especially for null check. Something like this:
```kotlin
inline fun <T, P : Any> T.runIfNotNull(parameter: P?, block: T.(P) -> T): T {
    return if (parameter != null) {
        block(parameter)
    } else {
        this
    }
}
```
I'm not sure that it's worth it. What do you think?

Reply:
Thought about it yesterday, too, but it seemed like overhead.
But since you also had such an idea - added `applyIfNotNull` / `applyIfNotNullElse`.